api/lib/opentelemetry/distributed_context/propagation/trace_parent.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+module OpenTelemetry
+  module DistributedContext
+    module Propagation
+      # A TraceParent is an implemenation of the W3C trace context specification
+      # {SpanContext}
+      class TraceParent
+        InvalidFormatError = Class.new(Error)
+        InvalidVersionError = Class.new(Error)
+        InvalidTraceIDError = Class.new(Error)
+        InvalidSpanIDError = Class.new(Error)
+
+        TRACE_PARENT_HEADER = 'traceparent'
+        SUPPORTED_VERSION = 0
+        private_constant :SUPPORTED_VERSION
+        MAX_VERSION = 254
+        private_constant :MAX_VERSION
+
+        REGEXP = /^(?<version>[A-Fa-f0-9]{2})-(?<trace_id>[A-Fa-f0-9]{32})-(?<span_id>[A-Fa-f0-9]{16})-(?<flags>[A-Fa-f0-9]{2})(?<ignored>-.*)?$/.freeze
+        private_constant :REGEXP
+
+        attr_reader :version, :trace_id, :span_id, :flags
+
+        private_class_method :new
+
+        def sampled?
+          flags.sampled?
+        end
+
+        def to_s
+          "00-#{trace_id}-#{span_id}-#{flags}"
+        end
+
+        def self.from_context(ctx)
+          new(trace_id: ctx.trace_id, span_id: ctx.span_id, flags: ctx.trace_flags)
+        end
+
+        def self.from_string(string)
```

Comment:
Would be good to document all the public methods here. Especially the ones that `raise`.